components/brave_wallet_ui/page/screens/send/send/send.tsx

```diff
@@ -344,6 +361,43 @@ export const Send = (props: Props) => {
     }
   }, [selectedTokensNetwork])
 
+  React.useEffect(() => {
+    // check if the user has selected an asset
+    if (!contractAddress || !accountAddress || !chainId || selectedSendAsset) return
+
+    const asset = sendAssetOptions.find((option) =>
+      tokenId
+        ? option.contractAddress.toLowerCase() ===
+            contractAddress.toLowerCase() && option.tokenId === tokenId
+        : option.contractAddress.toLowerCase() === contractAddress.toLowerCase()
+    )
```

Comment:
should `selectedSendAsset` be included in this computation?

Reply:
@josheleonard Yes, we should not override what the user has selected. This can happen when a user selects an asset and the component rerenders, it would trigger the useEffect and that would override their choice. That is why the check is necessary.

Reply:
I don't think that is necessary.